src/test/ui/mismatched_types/closure-arg-type-mismatch.stderr

```diff
@@ -0,0 +1,36 @@
+error[E0622]: type mismatch in closure arguments
+  --> $DIR/closure-arg-type-mismatch.rs:15:26
+   |
+15 |         let b = a.iter().map(|x: (u32, u32)| 45);
+   |                          ^^^ ------------------ takes a `std::ops::FnMut<((u32, u32),)>`
+   |                          |
+   |                          expected closure that takes a `std::ops::FnMut<(&(u32, u32),)>`
```

Comment:
I'm thinking about implementing a hint telling the user to add a `&`/`&mut` to the closure